pkg/jobs/utils.go

```diff
@@ -73,3 +75,46 @@ ORDER BY created`
 	}
 	return false /* exists */, err
 }
+
+// JobExists checks that whether there are any job records which exist that match
+// the payloadPredicate callback.
+func JobExists(
```

Comment:
how is this different from RunningJobExists above?
Purely based on status predicate?
Maybe we can avoid code duplication?

Reply:
Ack. I combined them into one function.